### PR TITLE
To verify that the cluster remains accessible and NO DU/DL after failures

### PR DESCRIPTION
**Pull request by Akarsha-rai**

### CODE REVIEW

File: tests/functional/disaster-recovery/metro-dr/test_no_data_loss_and_corruption_on_failures.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    1) Noobaa pods failures - repeat at least 5-7 times
+    2) Rolling reboot of the nodes in all zones one at a time
+    3) RHCS nodes failures
+        a. 1 RHCS node in one zone
+        b. All the RHCS nodes in one zone
+        c. All the RHCS nodes in one zone - Repeated to mimic Santa lucia issue
+
+    """
+
+    @pytest.mark.polarion_id("OCS-XXXX")
```
Comment (hnallurv) [inline tests/functional/disaster-recovery/metro-dr/test_no_data_loss_and_corruption_on_failures.py:41]: please add polarion id. Also, add this id to this PR heading/summary.

File: tests/functional/disaster-recovery/metro-dr/test_no_data_loss_and_corruption_on_failures.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+            namespace=self.namespace
+        )
+
+        # Validate data integrity
+        for wl in workloads:
+            config.switch_to_cluster_by_name(self.primary_cluster_name)
+            validate_data_integrity(wl.workload_namespace)
+
+        # Noobaa pod restarts atleast 5 times and verify the data integrity
+        restart_pods_having_label(label=constants.NOOBAA_APP_LABEL)
```
Comment (hnallurv) [inline tests/functional/disaster-recovery/metro-dr/test_no_data_loss_and_corruption_on_failures.py:69]: IIUC, this line does restart only once. we need 5 times. Isn't loop missing here?

File: tests/functional/disaster-recovery/metro-dr/test_no_data_loss_and_corruption_on_failures.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        for wl in workloads:
+            config.switch_to_cluster_by_name(self.primary_cluster_name)
+            validate_data_integrity(wl.workload_namespace)
+
+        # Get the nodes from one active zone
```
Comment (hnallurv) [inline tests/functional/disaster-recovery/metro-dr/test_no_data_loss_and_corruption_on_failures.py:74]: Elaborate the comment to mention what failure this code block is trying to achieve

File: tests/functional/disaster-recovery/metro-dr/test_no_data_loss_and_corruption_on_failures.py [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+        self, setup_acm_ui, nodes_multicluster, dr_workload
+    ):
+
+        # Deploy Subscription based application
+        sub = dr_workload(num_of_subscription=1)[0]
+        self.namespace = sub.workload_namespace
+        self.workload_type = sub.workload_type
+
+        # Deploy AppSet based application
+        appset = dr_workload(num_of_subscription=0, num_of_appset=1)[0]
```
Comment (parikshithb) [inline tests/functional/disaster-recovery/metro-dr/test_no_data_loss_and_corruption_on_failures.py:52]: Cant we deploy sub and appset apps using single call?

**Comment by Akarsha-rai** [comment]
Test Passed: https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-mdr-setup/559/testReport/

File: tests/functional/disaster-recovery/metro-dr/test_no_data_loss_and_corruption_on_failures.py (hunk 4 above)
Comment (Akarsha-rai) [reply tests/functional/disaster-recovery/metro-dr/test_no_data_loss_and_corruption_on_failures.py]: done

File: tests/functional/disaster-recovery/metro-dr/test_no_data_loss_and_corruption_on_failures.py (hunk 3 above)
Comment (Akarsha-rai) [reply tests/functional/disaster-recovery/metro-dr/test_no_data_loss_and_corruption_on_failures.py]: done

File: tests/functional/disaster-recovery/metro-dr/test_no_data_loss_and_corruption_on_failures.py (hunk 2 above)
Comment (Akarsha-rai) [reply tests/functional/disaster-recovery/metro-dr/test_no_data_loss_and_corruption_on_failures.py]: yes, right. Added now

File: tests/functional/disaster-recovery/metro-dr/test_no_data_loss_and_corruption_on_failures.py (hunk 1 above)
Comment (Akarsha-rai) [reply tests/functional/disaster-recovery/metro-dr/test_no_data_loss_and_corruption_on_failures.py]: added

**Review by hnallurv — DISMISSED** [review]

**Review by parikshithb — DISMISSED** [review]
lgtm

File: tests/functional/disaster-recovery/metro-dr/test_no_data_loss_and_corruption_on_failures.py [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+)
+from ocs_ci.framework.pytest_customization.marks import turquoise_squad
+from ocs_ci.utility import vsphere
+from ocs_ci.utility.utils import ceph_health_check
+
+logger = logging.getLogger(__name__)
+
+
+@tier4a
+@turquoise_squad
```
Comment (prsurve) [inline tests/functional/disaster-recovery/metro-dr/test_no_data_loss_and_corruption_on_failures.py:26]: need vsphere only marker

Comment (Akarsha-rai) [reply tests/functional/disaster-recovery/metro-dr/test_no_data_loss_and_corruption_on_failures.py:29]: thank you, added now

**Review by prsurve — APPROVED** [review]

**Review by parikshithb — APPROVED** [review]

**Review by vavuthu — APPROVED** [review]

**Pull request merged**